test/string_empty_leading_nul.rb

```
@@ -0,0 +1,31 @@
+bin = [0x00, 0x01, 0x02].pack("C*")
+p [bin.bytesize, bin.empty?]
+p [bin.byteslice(0, 1).bytesize, bin.byteslice(0, 1).empty?]
+
+# The answer has to agree with the length, whichever way it comes out.
+[".", "\0", "\0\0", "", "a\0b"].each do |s|
+  p [s.bytesize, s.empty?, s.bytesize == 0]
+end
+
+# A leading NUL through a variable and through a method call, so the answer is
+# not a constant folded from the literal.
+def first_nul(s) = "\0" + s
```

RuboCop (1.89.0)

[fatal] unexpected token tEQL
(Using Ruby 2.7 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

(Lint/Syntax)